### PR TITLE
[ENH] Add `pyarrow` and `s3fs`

### PR DESCRIPTION
**Pull request by viniciusdc**
Related to https://github.com/Quansight/qhub/issues/1359

## Changes introduced in this PR:

- Add two new packages to Dask env, `pyarrow` and `s3fs`

## Types of changes

What types of changes does your PR introduce?

_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

### Requires testing

- [ ] Yes
- [ ] No

### In case you checked yes, did you write tests?

- [ ] Yes
- [ ] No

## Documentation

Does your contribution include breaking changes or deprecations?
If so have you updated the documentation?

- [ ] Yes, docstrings
- [ ] Yes, main documentation
- [ ] Yes, deprecation notices

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered and more.

### CODE REVIEW

**Review by iameskild — APPROVED** [review]
Looks good to me :)

I wonder if it's also worth adding `gcsfs`, what do you think?

**Comment by viniciusdc** [comment]
> Looks good to me :)
> 
> I wonder if it's also worth adding `gcsfs`, what do you think?

If we add `s3sf` I don't see a problem, though to be honest I would prefer to keep `s3fs` and `gcsfs` as something the user should add to a personal env based on `filesystem/dask`. The reason is, that dask should be the basic env for running dask operations, anything outside this should go to a user/group-based env.

File: qhub/initialize.py [hunk 1]
```diff
@@ -241,6 +241,8 @@
             "ipywidgets",
             f"qhub-dask =={qhub_dask_version}",
             "python-graphviz",
+            "pyarrow",
+            "s3fs",
```
Comment (viniciusdc) [inline qhub/initialize.py:245]:
```suggestion
            "s3fs",
            "gcsf",
```

File: qhub/initialize.py [hunk 2]
```diff
@@ -241,6 +241,9 @@
...
+            "gcsf",
```
Comment (viniciusdc) [inline qhub/initialize.py:246]:
```suggestion
            "gcsfs",
```

**Review by iameskild — APPROVED** [review]

**Pull request merged**